### PR TITLE
fix: gfortran linux build

### LINKED ISSUE #12612: [package] gfortran/10.2: source code download url broken?
### Package and Environment Details

* Package Name/Version: **gfortran/10.2**
* Operating System+version: **Linux Ubuntu 18.04**
* Compiler+version: **GCC 7**
* Conan version: **conan 1.18.0**

### Conan profile

```
Configuration:
[settings]
arch=x86_64
arch.march=None
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=7
os=Linux
os_build=Linux
[options]
[build_requires]
[env]
```

### Steps to reproduce

I just tried to build the gfortran package locally
```
conan create recipes/gfrotran/all/ 10.2@_/_
```

### Logs

<details><summary>Click to expand log</summary>

```
gfortran/10.2: Forced build from source
gfortran/10.2 (test package): Installing package
Requirements
    gfortran/10.2 from local cache - Cache
Packages
    gfortran/10.2:4db1be536558d833e52e862fd84d64d75c2b3656 - Build

Installing (downloading, building) binaries...
gfortran/10.2: Configuring sources in /home/g.gulgulia/.conan/data/gfortran/10.2/_/_/source
ERROR: gfortran/10.2: Error in source() method, line 44
        tools.get(**url[it])
        NotFoundException: Not found: https://gfortran.meteodat.ch/download/x86_64/snapshots/gcc-10-20210109.tar.xz
```

</details>

### PR DESCRIPTION
**Pull request by ggulgulia**
Specify library name and version:  **gfortran/10.2**

fixes #12612

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ggulgulia** [comment]
@prince-chrismc  not sure why I get the error on conandata.yml ? Any suggestions to fix the hooks error ?

**Comment by uilianries** [comment]
@ggulgulia please, put urls and sha256 values as string, double quoted, like:

```
"Macos":
   url: "https://downloads.sourceforge.net/project/hpc/hpc/g95/gfortran-10.2-bin.tar.gz"
   sha256: "2de46f571eefc2b544db4ed6c958c78a5e3e78c4b4b5daab38aabc1b08dd5666"
```

It should fix that error.

**Comment by ggulgulia** [comment]
@uilianries  thanks for the reply. 

I did it in [this](https://github.com/conan-io/conan-center-index/pull/12613/commits/7c5a45ffec0ec313c4c61178406983003a75beff) commit but it did not fix the error. 

Let me try again.

**Comment by ggulgulia** [comment]
@uilianries  the error 
```
ERROR: [CONANDATA.YML FORMAT (KB-H030)] Additional entries ['filename'] not allowed in 'sources':'10.2' of conandata.yml 
```
is bit difficult for me to fix since I don't have access to a windows machine. Do you remember why `filename` entry was used in the `conandata.yml` in the original recipe ?

**Comment by prince-chrismc** [comment]
You should be able to clean up the `source()` with https://github.com/conan-io/conan-center-index/blob/8ebee1b90419165455095b5d92d3e5dbef770a8f/recipes/fmt/all/conanfile.py#L78

seems like this was not updated in a long time and needs a bunch of TLC

**Comment by ggulgulia** [comment]
@

> You should be able to clean up the `source()` with
> 
> https://github.com/conan-io/conan-center-index/blob/8ebee1b90419165455095b5d92d3e5dbef770a8f/recipes/fmt/all/conanfile.py#L78
> 
> seems like this was not updated in a long time and needs a bunch of TLC

@prince-chrismc  this doesn't work with the structure of conandata.yml .

**Comment by prince-chrismc** [comment]
> this doesn't work with the structure of conandata.yml .

Yep sorry for the bad reference https://github.com/conan-io/conan-center-index/blob/master/recipes/android-ndk/all/conanfile.py#L64-L65

Is close but not good enough since the mignw needs 7z... no way to do that easily.

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
You'll need to retrigger CI, close the pr wait 10s and then re-open it 🔁

I think you pushed during the service outage 🙊

File: recipes/gfortran/all/conanfile.py [hunk 1]
```diff
@@ -34,21 +34,8 @@ def build_requirements(self):
...
-                self.run("7z x {0}".format(filename))
-                os.unlink(filename)
-                os.rename("mingw64", "source_subfolder_Windows")
-            elif it != "Windows":
-                get(self, ** data[it]["x86_64"])
-                pattern = "gcc-*" if it == "Linux" else "usr"
-                os.rename(glob.glob(pattern)[0], "source_subfolder_{}".format(it))
-                break
+        get(self, **self.conan_data["sources"][self.version][str(self.settings.os)]["x86_64"],
+                  destination=self.source_folder, strip_root=True)
```
Comment (prince-chrismc) [inline recipes/gfortran/all/conanfile.py:38]: Should have been

```suggestion
                  destination=self._source_subfolder, strip_root=True)
```

**Comment by prince-chrismc** [comment]
You also need to bump the require version to 1.33.0 🙏

**Comment by prince-chrismc** [comment]
(misclicked) you see nothing

File: recipes/gfortran/all/conanfile.py [hunk 2]
```diff
@@ -32,23 +34,14 @@ def build_requirements(self):
...
-                tools.get(**url[it])
-                pattern = "gcc-*" if it == "Linux" else "usr"
-                os.rename(glob.glob(pattern)[0], "source_subfolder_{}".format(it))
+        get(self, **self.conan_data["sources"][self.version][str(self.settings.os)]["x86_64"],
+                  destination=self.source_folder, strip_root=True)
+
 
     def _extract_license(self):
-        info = tools.load(os.path.join(self.source_folder, "source_subfolder_Linux", "share", "info", "gfortran.info"))
+        info = load(self, os.path.join(self.source_folder, "source_subfolder_Linux", "share", "info", "gfortran.info"))
```
Comment (prince-chrismc) [inline recipes/gfortran/all/conanfile.py:42]: Should be using the property... idk why its different 😕

**Comment by ggulgulia** [comment]
@prince-chrismc  any idea why CI is taking so long for Macos build ?

**Comment by prince-chrismc** [comment]
Please try  not to use CCI to test 🙏

**Comment by ggulgulia** [comment]
> Please try  not to use CCI to test 🙏 

Sorry for that. I'M havng trouble getting the  windows build to work due to lack of windows system at hand. Do you know what could cause the current error in build?

**Comment by prince-chrismc** [comment]
Seems like the file paths are bad. 🤔 

When I get back home from CppCon I'll try to see if I can get this.

Hopefully someone else can chime in.

File: recipes/gfortran/all/conanfile.py [hunk 3]
```diff
@@ -32,23 +31,18 @@ def build_requirements(self):
...
-                tools.download(**url[it])
-                self.run("7z x {0}".format(filename))
-                os.unlink(filename)
-                os.rename("mingw64", "source_subfolder_Windows")
-            elif it != "Windows":
-                tools.get(**url[it])
-                pattern = "gcc-*" if it == "Linux" else "usr"
-                os.rename(glob.glob(pattern)[0], "source_subfolder_{}".format(it))
+        #does not work for windows ?
+        get(self, **self.conan_data["sources"][self.version][str(self.settings.os)]["x86_64"],
```
[the remark this answers is not in the document]
Comment (ggulgulia) [reply recipes/gfortran/all/conanfile.py:38]: done. Lets see if this fixes the build. Thanks for suggestions though

**Comment by prince-chrismc** [comment]
:wave: I am trying to open a PR against your fork! but I forgot how

**Comment by prince-chrismc** [comment]
If you revert your commit you can take in https://github.com/ggulgulia/conan-center-index/pull/3

**Comment by prince-chrismc** [comment]
Also please do not use CCI for testing 🙈 the downloads are actually pretty slow

**Comment by ggulgulia** [comment]
> Also please do not use CCI for testing see_no_evil the downloads are actually pretty slow

I am facing challenges from India to download the zip/tar files due to very slow internet connections. It will be great it you guys can take over the PR and continue the work. :pray:

**Review by danimtb — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
I'm in the USA and the download speeds where trash so it's not just you 😆 

Should be merged soon 👍
